### PR TITLE
SB-1227 LDAP Controller improvements

### PR DESCRIPTION
**Pull request by fuss86**

### CODE REVIEW

File: strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/registry/AuthenticatorsRegistry.java [hunk 1]
```diff
@@ -8,10 +8,15 @@
...
 import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Iterator;
+import java.util.List;
 import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import edu.emory.mathcs.backport.java.util.Collections;
```
Comment (sbespalov) [inline strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/registry/AuthenticatorsRegistry.java]: wrong import

Comment (fuss86) [reply strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/registry/AuthenticatorsRegistry.java]: Thanks

File: strongbox-security/strongbox-authentication-providers/strongbox-ldap-authentication-provider/src/main/java/org/carlspring/strongbox/authentication/external/ldap/LdapConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+            container.setPort(Integer.valueOf(port));
+            container.afterPropertiesSet();
+        }
+        catch (Exception ex)
+        {
+            throw Throwables.propagate(ex);
+        }
+        registerSingleton(applicationContext, container);
+
+        DefaultSpringSecurityContextSource contextSource = new DefaultSpringSecurityContextSource("ldap://127.0.0.1:" +
```
Comment (carlspring) [inline strongbox-security/strongbox-authentication-providers/strongbox-ldap-authentication-provider/src/main/java/org/carlspring/strongbox/authentication/external/ldap/LdapConfiguration.java]: Uhm... What is this about and why is it hardcoded?

Comment (fuss86) [reply strongbox-security/strongbox-authentication-providers/strongbox-ldap-authentication-provider/src/main/java/org/carlspring/strongbox/authentication/external/ldap/LdapConfiguration.java]: Yes, but this is how the Spring solution also works (as pointed above: `org.springframework.security.config.ldap.LdapServerBeanDefinitionParser#createEmbeddedServer`)

File: strongbox-web-core/src/test/resources/etc/conf/strongbox.xml [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+
+    <session-configuration timeout-seconds="3600" />
+
+    <remote-repositories-configuration check-interval-seconds="60" heartbeat-threads-number="5">
+        <retry-artifact-download-configuration timeout-seconds="60"
+                                               max-number-of-attempts="5"
+                                               min-attempts-interval-seconds="5"/>
+    </remote-repositories-configuration>
+
+    <cors-configuration allowed-credentials="true" max-age="600">
```
Comment (carlspring) [inline strongbox-web-core/src/test/resources/etc/conf/strongbox.xml]: I have the feeling that this configuration should be extracted to one of the other `XML` configuration files. Perhaps we should add it as a follow-up task...?

File: strongbox-web-core/src/test/resources/etc/conf/strongbox.xml [hunk 4]
```diff
@@ -0,0 +1,387 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (carlspring) [inline strongbox-web-core/src/test/resources/etc/conf/strongbox.xml]: Is there something special about this configuration file, that is different than the standard `strongbox.xml` file? (It's a bit hard to tell, so I'm just asking for my better understanding).

**Comment by carlspring** [comment]
Thanks for this EPIC pull and all the hard work! :)

**Pull request merged**